static/js/components/VideoPlayer.js

```diff
@@ -1,37 +1,225 @@
-          className="uswitch-odl-medium"
-          frameBorder="0"
-          scrolling="no"
-          allowFullScreen={true}
-        />;
-      } else {
-        return <USwitchPlayer video={video} />;
+  updateSubtitles() {
+    const { video } = this.props;
+    if (this.player) {
+      // Remove existing tracks for deleted subtitles
+      const tracks = this.player.textTracks();
+      let subtitleUrls = video.videosubtitle_set.map((subtitle: VideoSubtitle) => (makeVideoSubtitleUrl(subtitle)));
+      let trackUrls = [];
+      for (let idx = 0; idx < tracks.length; idx++) {
+        if (this.player && tracks[idx] && !subtitleUrls.includes(tracks[idx].src)) {
+          this.player.removeRemoteTextTrack(tracks[idx]);
+        } else {
+          trackUrls.push(tracks[idx].src);
+        }
       }
+      // Add tracks for any new subtitles associated with the video
+      video.videosubtitle_set.map((subtitle: VideoSubtitle) => {
+        const subUrl = makeVideoSubtitleUrl(subtitle);
+        if (!trackUrls.includes(subUrl)) {
+          this.player.addRemoteTextTrack({
+            kind: "captions",
+            src: subUrl,
+            srcLang: subtitle.language,
+            label: subtitle.language_name
+          });
+        }
+      });
+    }
+  }
+
+  drawCanvas(canvas: HTMLCanvasElement, shiftX: boolean, shiftY: boolean) {
+    function drawCanvasImage(canvas, videoNode, shiftX, shiftY) {
+      const x = shiftX ? Math.floor(videoNode.videoWidth / 2) : 0;
+      const y = shiftY ? Math.floor(videoNode.videoHeight / 2) : 0;
+      const context = canvas.getContext('2d');
+      context.drawImage(
+        videoNode,
+        x, y, Math.floor(videoNode.videoWidth / 2), Math.floor(videoNode.videoHeight / 2),
+        0, 0, canvas.width, canvas.height
+      );
+      setTimeout(drawCanvasImage, 20, canvas, videoNode, shiftX, shiftY);
+    }
+
+    canvas.width = Math.floor(this.videoNode.offsetWidth / 4) - 2;
+    canvas.height = Math.floor(this.videoNode.offsetHeight / 4) - 2;
+    if (canvas && this.videoNode) {
+      drawCanvasImage(canvas, this.videoNode, shiftX, shiftY);
+    }
+  }
+
+  configureCameras() {
+    if (this.cameras) {
+      const canvasElements = this.cameras.getElementsByTagName("canvas");
+      Object.keys(canvases).map(corner => {
+        this.drawCanvas(
+          canvasElements[corner],
+          canvases[corner].shiftX,
+          canvases[corner].shiftY
+        );
+      });
+    }
+  }
+
+  isFullscreen() {
+    // $FlowFixMe
+    return document[FULLSCREEN_API.fullscreenElement];
+  }
+
+  cropVideo() {
+    const { selectedCorner } = this.props;
+    if (selectedCorner) {
+      const shiftX = canvases[selectedCorner].shiftX;
+      const shiftY = canvases[selectedCorner].shiftY;
+      const transformProps = ['transform', 'WebkitTransform', 'MozTransform', 'msTransform', 'OTransform'];
+      let prop = transformProps[0];
+
+      transformProps.some((property: string) => {
```

Comment:
Can you change this to a `find` to be more explicit: `prop = transformProps.find(property => this.player.el_.style[prop] !== undefined)`?

Also the `typeof variable === 'undefined'` is not necessary anymore because since ES5 it always exists and cannot be overridden: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/undefined

Reply:
OK